### PR TITLE
SUR-248 Implement a Container Component 

### PR DESCRIPTION
**Pull request by JelenaTakac**
### Description

# Container

## Description

The `Container` component is a flexible layout component that can be used to create grid or flexbox-based layouts.

### containerType
- **Type:** `string`
- **Default:** `"grid"`
- **Description:** Specifies the type of container
    - `"flex"`
    - `"grid"`

### gap
- **Type:** `string`
- **Description:** Defines the gap between grid or flex items.
    - `"xs"`
    - `"sm"`
    - `"md"`
    - `"lg"`
    - `"xl"`
    - `"2xl"`

### direction
- **Type:** `string`
- **Default:** `"row"`
- **Description:** Defines the direction of flex items (applicable only for flex containers).
    - `"row"`
    - `"row-reverse"`
    - `"column"`
    - `"column-reverse"`

### justify
- **Type:** `string`
- **Default:** `"start"`
- **Description:** Specifies how flex items are aligned along the main axis.
    - `"normal"`
    - `"start"`
    - `"end"`
    - `"center"`
    - `"between"`
    - `"around"`
    - `"evenly"`
    - `"stretch"`

### align
- **Type:** `string`
- **Default:** `"stretch"`
- **Description:** Specifies how flex items are aligned along the cross axis.
    - `"start"`
    - `"end"`
    - `"center"`
    - `"baseline"`
    - `"stretch"`

### wrap
- **Type:** `string`
- **Default:** `nowrap`
- **Description:** Defines how flex items should wrap in the container.
    - `"nowrap"`
    - `"wrap"`
    - `"wrap-reverse"`

### className
- **Type:** `string`
- **Description:** Additional custom classes to be added to the container.

### children
- **Type:** `ReactNode`
- **Description:** The content to be rendered inside the container.

### extraProps
- **Type:** `object`
- **Description:** Additional props to be passed to the container's root element.

### grow
- **Type:** `number`
- **Description:** Specifies how much a flex item should grow relative to the rest of the flex items.
    - `"0"`
    - `"1"`

### shrink
- **Type:** `number`
- **Description:** Specifies how much a flex item should shrink relative to the rest of the flex items.
    - `"0"`
    - `"1"`

### order
- **Type:** `string`
- **Description:** Defines the order of the flex item in the container.
    - `"1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12"`
    - `"first"`
    - `"last"`
    - `"none"`

### alignSelf
- **Type:** `string`
- **Description:** Allows the default alignment (or the one specified by align-items) to be overridden for individual flex items.
    - `"auto"`
    - `"start"`
    - `"end"`
    - `"stretch"`
    - `"baseline"`

### Basic Example

```jsx
import { Container } from '@bsf/force-ui';

const App = () => (
    <Container containerType="flex" justify="end" align="start" gap="xs" className="bg-orange-300 h-60">
        <Container.Item grow={1}>Item 1</Container.Item>
        <Container.Item>Item 2</Container.Item>
        <Container.Item grow={1}>Item 3</Container.Item>
    </Container>
);

export default App;
```

### Checklist:

-   [ ] My code is tested
-   [ ] My code passes the PHPCS tests
-   [ ] I've created the npm build.
-   [ ] My code follows accessibility standards 
-   [ ] My code has proper inline documentation 
-   [ ] I've included any necessary tests 
-   [ ] I've included developer documentation 
-   [ ] I've added proper labels to this pull request

### CODE REVIEW

**Comment by vrundakansara** [comment]
@JelenaTakac - Fix the lint pipeline by running `npm run lint:js-fix`.
Also push the code with the build - `npm run build`.

File: src/components/container/container.jsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { cn } from '../../utilities/functions';
+
+const Container = ( props ) => {
+	const {
+		containerType = 'grid', // flex, grid
```
Comment (vrundakansara) [inline src/components/container/container.jsx:6]: Default should be flex

File: src/components/container/container.jsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		gap, // xs, sm, md, lg, xl, 2xl
```
Comment (vrundakansara) [inline src/components/container/container.jsx:7]: Keep `sm` as default

File: src/components/container/container.jsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		direction = 'row', // row, row-reverse, column, column reverse
+		justify = 'start', // justify-content (normal, start, end, center, between, around, evenly, stretch)
+		align = 'stretch', // align-items (start, end, center, baseline, stretch)
```
Comment (vrundakansara) [inline src/components/container/container.jsx:10]: Set it to blank as default

File: src/components/container/container.jsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { cn } from '../../utilities/functions';
+
+const Container = ( props ) => {
+	const {
+		containerType = 'grid', // flex, grid
+		gap, // xs, sm, md, lg, xl, 2xl
+		direction = 'row', // row, row-reverse, column, column reverse
+		justify = 'start', // justify-content (normal, start, end, center, between, around, evenly, stretch)
```
Comment (vrundakansara) [inline src/components/container/container.jsx:9]: Blank as default

File: src/components/container/container.jsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { cn } from '../../utilities/functions';
+
+const Container = ( props ) => {
+	const {
+		containerType = 'grid', // flex, grid
+		gap, // xs, sm, md, lg, xl, 2xl
+		direction = 'row', // row, row-reverse, column, column reverse
```
Comment (vrundakansara) [inline src/components/container/container.jsx:8]: Blank as default

**Review by vrundakansara — CHANGES_REQUESTED** [review]

File: src/components/container/container.jsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+	const {
+		containerType = 'flex', // flex, grid
+		gap = 'sm', // xs, sm, md, lg, xl, 2xl
+		direction = '', // row, row-reverse, column, column reverse
+		justify = '', // justify-content (normal, start, end, center, between, around, evenly, stretch)
+		align = '', // align-items (start, end, center, baseline, stretch)
+		wrap = '', // nowrap, wrap, wrap-reverse
+		className,
+		children,
+		extraProps,
```
Comment (jaieds) [inline src/components/container/container.jsx:14]:
```suggestion
		...extraProps,
```

File: dist/utils/withTW.js
Comment (vrundakansara) [inline dist/utils/withTW.js:1]: Any reason why we deleted this file?

**Review by vrundakansara — CHANGES_REQUESTED** [review]

File: dist/utils/withTW.js
Comment (JelenaTakac) [reply dist/utils/withTW.js:1]: @vrundakansara, can you please check it now?

**Review by vrundakansara — APPROVED** [review]

**Review by jaieds — APPROVED** [review]

**Pull request merged**